src/data/mv_polynomial/basic.lean

```diff
@@ -811,6 +832,15 @@ begin
   exact hf hx
 end
 
+lemma C_dvd_iff_map_hom_eq_zero
+  (q : α →+* β) (r : α) (hr : ∀ r' : α, q r' = 0 ↔ r ∣ r')
+  (φ : mv_polynomial σ α) :
```

Comment:
Some of these variables can be implicit, right?

Reply:
In principle yes, but typically you won't have a proof of `hr` lying around, and you want it to become a new goal.
So then it helps if you can just specify `q` and `r` without having to use `@`.